cla_backend/apps/checker/models.py

```diff
@@ -233,21 +233,64 @@ class CallbackTimeSlot(TimeStampedModel):
 
     @property
     def remaining_capacity(self):
+        return self.get_remaining_capacity_by_range(
+            self.capacity, self.callback_start_datetime(), self.callback_end_datetime()
+        )
+
+    @staticmethod
+    def get_model_from_datetime(dt, fallback_to_previous_week=True):
+        assert isinstance(dt, datetime.datetime)
+        is_fallback = False
+        if timezone.is_naive(dt):
+            dt = timezone.make_aware(dt)
+        model = CallbackTimeSlot.objects.filter(date=dt.date(), time=dt.strftime("%H%M")).first()
+
+        if not model and fallback_to_previous_week:
+            previous_week = dt - datetime.timedelta(weeks=1)
+            model = CallbackTimeSlot.objects.filter(
+                date=previous_week.date(), time=previous_week.strftime("%H%M")
+            ).first()
+            is_fallback = True
+
+        return is_fallback, model
+
+    @staticmethod
+    def get_time_from_interval_string(interval):
+        hour = int(interval[0:2])
+        minutes = int(interval[2:])
+        return datetime.time(hour=hour, minute=minutes)
+
+    @staticmethod
+    def get_remaining_capacity_by_dt(capacity, dt):
+        start = dt
+        end = dt + datetime.timedelta(minutes=SLOT_INTERVAL_MINS)
+        return CallbackTimeSlot.get_remaining_capacity_by_range(capacity, start, end)
+
+    @staticmethod
+    def get_remaining_capacity_by_date(dt):
```

Comment:
Can we have a DocString explaining the purpose of this function and why 99,999 is returned if no callback capacity exists?